### DISCUSSION TITLE
anvil doesn't work properly

### DISCUSSION THREAD

**Discussion by riyanparag1809** [Cyfrin/foundry-full-course-cu#2725] [Q&A]
forge create SimpleStorage --rpc-url http://127.0.0.1:7545/ --interactive
[⠊] Compiling...
No files changed, compilation skipped
Error:
error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:

Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
Error https://github.com/Cyfrin/foundry-full-course-cu/discussions/1: tcp connect error: Connection refused (os error 111)
Error https://github.com/Cyfrin/foundry-full-course-cu/discussions/2: Connection refused (os error 111)

**Comment by EngrPips** [comment]
Hello @riyanparag1809, Ensure that you are running an Anvil chain in  a separate terminal and you keep it running while running the `forge create SimpleStorage --rpc-url http://127.0.0.1:7545/ --interactive` command

**Reply by riyanparag1809** [reply]
Thank you 👍 I got it

**Reply by EngrPips** [reply]
Awesome, Keep crushing it, Friend.

**Reply by Vermitrude** [reply]
I encountered this same issue and was about to open a new discussion but opted to search if related questions has been asked. Let me try this out now.

**Reply by EngrPips** [reply]
Alright, I hope this help you fix it.

**Reply by EngrPips** [reply]
Hello @riyanparag1809, use `http://127.0.0.1:8545/` instead of `https://127.0.0.1:8545/`

**Reply by chainflash** [reply]
awesome

**Comment by chainflash** [comment]
hello buddy
Everything is correct, you just need to change the port number.
use the below command instead of your own command and then it will be fixed 💯 
forge create SimpleStorage --rpc-url http://127.0.0.1:8545
or 
forge create SimpleStorage --rpc-url http://127.0.0.1:8545 --interactive
and be sure you enter following code in another terminal and see Listening on 127.0.0.1:8545 in the output:
anvil --port 8545

**Reply by EngrPips** [reply]
Interesting.

**Comment by ServerSiri** [comment]
What if I am using ganache

**Reply by EngrPips** [reply]
Honestly, I will advise you to switch to Anvil.
